src/dotnet-install.ps1

```diff
@@ -131,6 +131,29 @@ function Say($str) {
     }
 }
 
+function Say-Warning($str) {
+    try
+    {
+        Write-Warning "dotnet-install: $str"
+    }
+    catch
+    {
+        # Some platforms cannot utilize Write-Warning (Azure Functions, for instance). Fall back to Write-Output
+        Write-Output "Warning: dotnet-install: $str"
+    }
+}
+
+function Say-Error($str) {
+    try
+    {
+        $Host.UI.WriteErrorLine("dotnet-install: $str")
+    }
+    catch
+    {
```

Comment:
minor: this pattern here and in `Say-Warning` looks "funny".

Reply:
I am also not sure why this is used over Write-Error or Write-Host, or throw. Would be nice to have a comment explaining it.

Reply:
This function's goal is to add a nice human-reading comment of what is going on before showing the exception.  Previously we threw these comments and lost the information in the exception (that's why not `throw`). This is basically the same as `Write-Host` but in a different color. (I know there are other ways to achieve that, but this is the one that I found. If there are ways to achieve that easier, I am ready to fix this). I found Write-Error being oververbose for this purpose.

Reply:
@donJoseLuis not sure what do you mean by "pattern". It is the same as in the function "Say" above. Could you please clarify?

Reply:
@AR-May that is the reasoning I would want to have in the code. Because I am sure it has its reason. It would just strike me as strange seeing that used, so comment would explain to me that this was a conscious choice.

Reply:
@nohwnd I see. Yes, let's put the comment about that.